### DISCUSSION TITLE
How to use $ref relative to the root?

### DISCUSSION THREAD

**Discussion by debuggerpk** [santhosh-tekuri/jsonschema#164] [Q&A] [answered]
Thank you for your awesome work.

I am looking to bump up the version to 4.0 for [cdevents](https://github.com/cdevents/spec/blob/v0.4.0/schemas/artifactdeleted.json#L41) for [sdk-go](https://github.com/cdevents/sdk-go).

When I pull the new version and run `go run ./tools/generator.go`, I get the error 

```txt
generator: Generating SDK files from templates: tools/templates/*.tmpl and schemas: ./pkg/api/spec/schemas into ./pkg/api
generator: Ignoring folder ./pkg/api/spec/schemas
generator: jsonschema file:///path-to-repo/cdevents/sdk-go/pkg/api/spec/schemas/artifactdeleted.json compilation failed: https://cdevents.dev/schema/links/embeddedlinksarray.json returned status code 404
exit status 1
```

On close inspection, version 4 has addition `"$ref": "/schema/links/embeddedlinksarray.json"`. I guess since `$id` is defined as `"$id": "https://cdevents.dev/0.4.0/schema/artifact-deleted-event"` and the schema is not published, we get the 404 error. Please note that we can find the `links` folder relative to `./pkg/api/spec/schemas`. Is the output _Ignoring folder a tell tale?_.

How to go about circumventing the aforementioned error.

**Comment by santhosh-tekuri** [comment, marked as answer]
You can use `Compiler.LoadURL` to handle this situation.

say you have cloned the schemas git repo into `/tmp`:
```
$ cd /tmp
$ git clone https://github.com/cdevents/spec.git
```

now you can compile schema as follows:

```go
prefix := "https://cdevents.dev/0.4.0/schema/"

c := jsonschema.NewCompiler()
c.LoadURL = func(s string) (io.ReadCloser, error) {
    suffix, ok := strings.CutPrefix(s, prefix)
    if !ok {
        return nil, fmt.Errorf("%s not found", s)
    }
   return os.Open("/tmp/"+suffix)
}
sch, err := c.Compile(prefix+"artifact-deleted-event")
```

**Reply by debuggerpk** [reply]
Thank you. Let me try.
